src/server/models/index.js

```diff
@@ -183,15 +196,39 @@ function aggregateESResult(params) {
     score: hit._score,
   }));
 
-  if (Object.keys(filter).length > 0) {
-    result.filters = filter;
-  }
-
   result.meta = meta;
   result.results = results;
   return result;
 }
 
+// called with every property and its value
+function enrichDocCounts(value, esAggregation) {
+  const currentValue = value;
+  const id = value.alt.dkultTermIdentifier;
+  const currentAggregation = esAggregation.filter((aggregation) => aggregation.display_value === id);
+  if (currentAggregation[0]) {
+    currentValue.doc_count = currentAggregation[0].doc_count;
+    currentValue.is_available = true;
+  }
+  else {
+    currentValue.doc_count = 0;
+    currentValue.is_available = false;
+  }
+}
+
+function traverse(obj, esAggregation, func) {
```

Comment:
Habe mal geschaut und ich glaube hier könnte man vllt. sogar `Object.values()` nutzen, um den Code etwas zu vereinfachen.

In etwas so dachte ich es mir:

```javascript
function traverse(obj, esAggregation, func) {
  Object.values(obj).forEach((value) => {
    func(value, esAggregation);

    const subTerms = (value || {}).subTerms;
    if (Array.isArray(subTerms)) {
      traverse(value.subTerms, esAggregation, func);
    }
  });
};
```

Im Code wird nun auch sicherheitshalber überprüft, ob `subTerms` überhaupt existiert und ein Array ist.
Würde zudem sagen, dass man das `apply()` hier nicht direkt braucht, da man in der `enrichDocCounts()` nicht auf den Gültigkeitsbereich der `traverse` zugreifen muss, was in diesem Falle wahrscheinlich maximal der Module-Scope (`module.exports`-Objekt) wäre.

Reply:
Okay, das sieht besser aus. Das mit dem Module-Scope musst du mir noch mal erklären. Das habe ich noch nicht gecheckt.

Reply:
Alles klar, jetzt habe ich es verstanden. Danke für die ausführliche Erläuterung :-)